ci/scripts/r_docker_configure.sh

```diff
@@ -77,5 +77,19 @@ if [ "$ARROW_S3" == "ON" ] || [ "$ARROW_R_DEV" == "TRUE" ]; then
   fi
 fi
 
+# Install patch if it doesn't already exist
+if [ ! $(command -v patch) ]; then
+  if [ "`which dnf`" ]; then
+    dnf install -y patch
+  elif [ "`which yum`" ]; then
+    yum install -y patch
+  elif [ "`which zypper`" ]; then
+    zypper install -y patch
+  else
+    apt-get update
+    apt-get install -y patch
+  fi
+fi
```

Comment:
This is not strictly necessary (see below where we disable snappy if we can't find `patch`), but it makes sure that we do have `patch` available so we do exercise this code in the sanitizer in our CI